src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1873,9 +1873,58 @@ void layout_optimizer::select_preferred_formats_for_onednn(program_node& node, d
             GPU_DEBUG_LOG << "select_preferred_formats:" << node.id() << ": " << fmt_to_str(target_format) << " --> " << fmt_to_str(target_format)
                           << " For index : " << idx << std::endl;
         }
+        // Optimized out permute from permute-gemm pattern. e.f permute -> gemm
+        if (node.is_type<gemm>()) {
+            for (size_t idx = 0 ; idx < node.get_dependencies().size() ; idx++) {
+                if (node.get_dependency(idx).is_type<permute>()) {
+                    auto& pnode = node.get_dependency(idx);
+                    if (pnode.has_fused_primitives()) {
+                        continue;
+                    }
+                    auto input_lay = pnode.get_dependency(0).get_output_layout();
+                    auto output_lay = pnode.get_output_layout();
+                    if (input_lay.compatible(output_lay)) {
```

Comment:
from my experience, there was a corner case where perf drop was observed. I think it is risky to merge any permute. How do you think?

Reply:
Let's check the performance model by mode, and discuss it